src/njs_object.c

```diff
@@ -917,7 +917,7 @@ njs_get_own_ordered_keys(njs_vm_t *vm, const njs_object_t *object,
     njs_lvlhsh_each_init(&lhe, &njs_object_hash_proto);
     hash = &object->shared_hash;
 
-    for ( ;; ) {
+    for ( ; !(flags & NJS_ENUM_NON_SHARED_ONLY); ) {
```

Comment:
this expression is const in the loop, better use explicit if before loop:

if (!(flags & NJS_ENUM_NON_SHARED_ONLY))
for ( ; ; ) {

Reply:
Will do a check with goto before. But I expect the effect of the change unmeasurable. Also wanted to reduce the diff, while preserving the style.